test/unit/Adapter/ParameterContainerTest.php

```diff
@@ -125,6 +125,15 @@ public function testSetFromArray()
     {
         $this->parameterContainer->setFromArray(['bar' => 'baz']);
         self::assertEquals('baz', $this->parameterContainer['bar']);
+        //Handle statement parameters - https://github.com/laminas/laminas-db/issues/47
+        //@see Insert::procesInsert as example
+        $this->parameterContainer->offsetSet('c_0', ':myparam');
+        $this->parameterContainer->setFromArray([':myparam' => 'baz']);
+        self::assertEquals('baz', $this->parameterContainer['c_0']);
+        self::assertEquals('baz', $this->parameterContainer[':myparam']);
+        $this->parameterContainer->setFromArray([':myparam' => 'baz2']);
+        self::assertEquals('baz2', $this->parameterContainer[':myparam']);
+        self::assertEquals('baz2', $this->parameterContainer['c_0']);
```

Comment:
I would suggest using a data supplier here, otherwise we will get repetitions.

Reply:
did you mean dataProvider ?

Reply:
I've removed the repetition, no need to loose time working on a dataprovider for that.

Reply:
@trasher please add signoff to your latest commit https://github.com/laminas/laminas-db/pull/160/checks?check_run_id=731302797

Reply:
@trasher
Please create a data provider because deleting two lines does not resolve the problem of repetition.

Thanks in advance!

Reply:
There are no repetition, two cases are different.

Reply:
I'm sorry, my explanation was too short. The entire test can be reduced to one call of the `setFromArray` method but with different values if a data provider is used.
Therefore I have asked for the modification.

Thanks for help!

Reply:
The test I've added is specific because there is a mandatory call to `offsetSet`, so this is different from the existing one (with no call to offsetset, which should still be tested I guess). So, they're different.

Therefore, to work with a dataprovider, it would require two different tests methods for each case, with 2 different data providers; ending in only one test case each. That's why I do not get the point.

Reply:
You extended an existing the test case for bugfix. Normally a new test must be defined to illustrate the wrong behaviour without to change other test. Now that you have already made the change, I have suggest a data provider to separate the test cases.

But you can also leave everything as it is, then it will be handled later.

Thanks for your help and time anyway! 👍

Reply:
OK, then I propose do simply do the following:
```diff
diff --git a/test/unit/Adapter/ParameterContainerTest.php b/test/unit/Adapter/ParameterContainerTest.php
index e06bb1fd..04ec0fdc 100644
--- a/test/unit/Adapter/ParameterContainerTest.php
+++ b/test/unit/Adapter/ParameterContainerTest.php
@@ -125,8 +125,17 @@ class ParameterContainerTest extends TestCase
     {
         $this->parameterContainer->setFromArray(['bar' => 'baz']);
         self::assertEquals('baz', $this->parameterContainer['bar']);
-        //Handle statement parameters - https://github.com/laminas/laminas-db/issues/47
-        //@see Insert::procesInsert as example
+    }
+
+    /**
+     *
+     * Handle statement parameters - https://github.com/laminas/laminas-db/issues/47
+     * @see Insert::procesInsert as example
+     *
+     * @covers \Laminas\Db\Adapter\ParameterContainer::setFromArray
+     */
+    public function testSetFromArrayNamed()
+    {
         $this->parameterContainer->offsetSet('c_0', ':myparam');
         $this->parameterContainer->setFromArray([':myparam' => 'baz']);
         self::assertEquals('baz', $this->parameterContainer['c_0']);
```

Does it sounds better to you?

Reply:
Go for it and we can add your PR to the next bugfix release. 👍